NebulaPatcher/Patches/Transpilers/UIProductEntry_Transpiler.cs

```diff
@@ -0,0 +1,126 @@
+﻿#region
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection.Emit;
+using HarmonyLib;
+using NebulaModel.Logger;
+using NebulaWorld;
+
+#endregion
+
+namespace NebulaPatcher.Patches.Transpilers;
+
+[HarmonyPatch(typeof(UIProductEntry))]
+public static class UIProductEntry_Transpiler
+{
+    [HarmonyTranspiler]
+    [HarmonyPatch(nameof(UIProductEntry.UpdateExtraProductTexts))]
+    private static IEnumerable<CodeInstruction> UpdateExtraProductTexts_Transpiler(IEnumerable<CodeInstruction> instructions)
+    {
+        //Change: this.gameData.factoryCount
+        //To:     GetFactoryCount()
+        //Change: planetData.factoryIndex
+        //To:     GetFactoryIndex(planetData)
+        //Change: planetData.factory != null
+        //To:     HasFactory(planetData)
+        var codeInstructions = instructions as CodeInstruction[] ?? instructions.ToArray();
+        try
+        {
+            instructions = ReplaceFactoryCount(codeInstructions);
+            instructions = new CodeMatcher(instructions)
+                .MatchForward(false,
+                    new CodeMatch(OpCodes.Ldfld, AccessTools.Field(typeof(PlanetData), nameof(PlanetData.factoryIndex)))
+                )
+                .Repeat(matcher => matcher
+                    .SetAndAdvance(OpCodes.Call,
+                        AccessTools.Method(typeof(UIProductEntry_Transpiler), nameof(GetFactoryIndex)))
+                )
+                .InstructionEnumeration();
+            /*
+            // planetData.factory != null => GetFactoryIndex(planetData) != -1
+            instructions = new CodeMatcher(instructions)
+                .MatchForward(false,
+                    new CodeMatch(OpCodes.Ldfld, AccessTools.Field(typeof(PlanetData), nameof(PlanetData.factory))),
+                    new CodeMatch(OpCodes.Brfalse)
+                )
+                .Repeat(matcher => matcher
+                    .SetAndAdvance(OpCodes.Call, AccessTools.Method(typeof(UIProductEntry_Transpiler), nameof(GetFactoryIndex)))
+                    .InsertAndAdvance(new CodeInstruction(OpCodes.Ldc_I4_M1))
+                    .SetOpcodeAndAdvance(OpCodes.Beq_S)
+                )
+                .InstructionEnumeration()
+            */
```

Comment:
Is this to be used in the future? Or leftover code that should be removed?

If needed in the future, adding a comment for it would be useful

Reply:
sorry, I forgot to remove them